commit 513389809e138ae903b6ef43c1d5d2ffaf4dca17
Merge: 0a78a376ef3c 30514bd2dd4e
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Fri Oct 7 09:19:14 2022 -0700

    Merge tag 'for-6.1/block-2022-10-03' of git://git.kernel.dk/linux
    
    Pull block updates from Jens Axboe:
    
     - NVMe pull requests via Christoph:
          - handle number of queue changes in the TCP and RDMA drivers
            (Daniel Wagner)
          - allow changing the number of queues in nvmet (Daniel Wagner)
          - also consider host_iface when checking ip options (Daniel
            Wagner)
          - don't map pages which can't come from HIGHMEM (Fabio M. De
            Francesco)
          - avoid unnecessary flush bios in nvmet (Guixin Liu)
          - shrink and better pack the nvme_iod structure (Keith Busch)
          - add comment for unaligned "fake" nqn (Linjun Bao)
          - print actual source IP address through sysfs "address" attr
            (Martin Belanger)
          - various cleanups (Jackie Liu, Wolfram Sang, Genjian Zhang)
          - handle effects after freeing the request (Keith Busch)
          - copy firmware_rev on each init (Keith Busch)
          - restrict management ioctls to admin (Keith Busch)
          - ensure subsystem reset is single threaded (Keith Busch)
          - report the actual number of tagset maps in nvme-pci (Keith
            Busch)
          - small fabrics authentication fixups (Christoph Hellwig)
          - add common code for tagset allocation and freeing (Christoph
            Hellwig)
          - stop using the request_queue in nvmet (Christoph Hellwig)
          - set min_align_mask before calculating max_hw_sectors (Rishabh
            Bhatnagar)
          - send a rediscover uevent when a persistent discovery controller
            reconnects (Sagi Grimberg)
          - misc nvmet-tcp fixes (Varun Prakash, zhenwei pi)
    
     - MD pull request via Song:
          - Various raid5 fix and clean up, by Logan Gunthorpe and David
            Sloan.
          - Raid10 performance optimization, by Yu Kuai.
    
     - sbitmap wakeup hang fixes (Hugh, Keith, Jan, Yu)
    
     - IO scheduler switching quisce fix (Keith)
    
     - s390/dasd block driver updates (Stefan)
    
     - support for recovery for the ublk driver (ZiyangZhang)
    
     - rnbd drivers fixes and updates (Guoqing, Santosh, ye, Christoph)
    
     - blk-mq and null_blk map fixes (Bart)
    
     - various bcache fixes (Coly, Jilin, Jules)
    
     - nbd signal hang fix (Shigeru)
    
     - block writeback throttling fix (Yu)
    
     - optimize the passthrough mapping handling (me)
    
     - prepare block cgroups to being gendisk based (Christoph)
    
     - get rid of an old PSI hack in the block layer, moving it to the
       callers instead where it belongs (Christoph)
    
     - blk-throttle fixes and cleanups (Yu)
    
     - misc fixes and cleanups (Liu Shixin, Liu Song, Miaohe, Pankaj,
       Ping-Xiang, Wolfram, Saurabh, Li Jinlin, Li Lei, Lin, Li zeming,
       Miaohe, Bart, Coly, Gaosheng
    
    * tag 'for-6.1/block-2022-10-03' of git://git.kernel.dk/linux: (162 commits)
      sbitmap: fix lockup while swapping
      block: add rationale for not using blk_mq_plug() when applicable
      block: adapt blk_mq_plug() to not plug for writes that require a zone lock
      s390/dasd: use blk_mq_alloc_disk
      blk-cgroup: don't update the blkg lookup hint in blkg_conf_prep
      nvmet: don't look at the request_queue in nvmet_bdev_set_limits
      nvmet: don't look at the request_queue in nvmet_bdev_zone_mgmt_emulate_all
      blk-mq: use quiesced elevator switch when reinitializing queues
      block: replace blk_queue_nowait with bdev_nowait
      nvme: remove nvme_ctrl_init_connect_q
      nvme-loop: use the tagset alloc/free helpers
      nvme-loop: store the generic nvme_ctrl in set->driver_data
      nvme-loop: initialize sqsize later
      nvme-fc: use the tagset alloc/free helpers
      nvme-fc: store the generic nvme_ctrl in set->driver_data
      nvme-fc: keep ctrl->sqsize in sync with opts->queue_size
      nvme-rdma: use the tagset alloc/free helpers
      nvme-rdma: store the generic nvme_ctrl in set->driver_data
      nvme-tcp: use the tagset alloc/free helpers
      nvme-tcp: store the generic nvme_ctrl in set->driver_data
      ...

diff --cc block/genhd.c
index 988ba52fd331,d6a21803a57e..514395361d7c
--- a/block/genhd.c
+++ b/block/genhd.c
@@@ -625,9 -626,7 +625,9 @@@ void del_gendisk(struct gendisk *disk
  	pm_runtime_set_memalloc_noio(disk_to_dev(disk), false);
  	device_del(disk_to_dev(disk));
  
 +	blk_mq_freeze_queue_wait(q);
 +
- 	blk_throtl_cancel_bios(disk->queue);
+ 	blk_throtl_cancel_bios(disk);
  
  	blk_sync_queue(q);
  	blk_flush_integrity();